lambda/index.py

```diff
@@ -0,0 +1,228 @@
+    # the record.
+    route_53_record_ttl=record_config_set['route_53_record_ttl']
+    route_53_record_type="A"
+    shared_secret=record_config_set['shared_secret']
+
+    # Validate that the client passed a sha256 hash
+    # regex checks for a 64 character hex string.
+    if not re.match(r'[0-9a-fA-F]{64}', validation_hash):
+        return_status='fail'
+        return_message='You must pass a valid sha256 hash in the '\
+            'hash= argument.'
+        return [400, {'return_status': return_status,
+                'return_message': return_message}]
+    # Calculate the validation hash.
+    hashcheck=source_ip + ddns_hostname + shared_secret
+    calculated_hash=hashlib.sha256(
+        hashcheck.encode('utf-8')).hexdigest()
+    # Compare the validation_hash from the client to the
+    # calculated_hash.
+    # If they don't match, error out.
+    if not calculated_hash == validation_hash:
+        return_status='fail'
+        return_message='Validation hashes do not match.'
+        return [401, {
+                'return_status': return_status,
+                'return_message': return_message}]
+    # If they do match, get the current ip address associated with
+    # the hostname DNS record from Route 53.
+    else:
+        route53_get_response=route53_client(
+            'get_record',
+            route_53_zone_id,
+            ddns_hostname,
+            route_53_record_ttl,
+            route_53_record_type,
+            '')
+        # If no records were found, route53_client returns null.
+        # Set route53_ip and stop evaluating the null response.
+        if route53_get_response['return_status'] == "fail":
+            return [500, route53_get_response]
+        else:
+            route53_ip = route53_get_response['return_message']
+        # If the client's current IP matches the current DNS record
+        # in Route 53 there is nothing left to do.
+        if route53_ip == source_ip:
+            return_status = 'success'
+            return_message = 'Your IP address matches '\
+                'the current Route53 DNS record.'
+            return [200, {'return_status': return_status,
+                    'return_message': return_message}]
+        # If the IP addresses do not match or if the record does not exist,
+        # Tell Route 53 to set the DNS record.
+        else:
+            return_status = route53_client(
+                'set_record',
+                route_53_zone_id,
+                ddns_hostname,
+                route_53_record_ttl,
+                route_53_record_type,
+                source_ip)
+            return return_status
+
+
+'''
+The function that Lambda executes. It contains the main script logic, calls and returns the output back to API Gateway
+'''
+
+
+def lambda_handler(event, context):
+    # Get execution mode and source IP
+    execution_mode = json.loads(event['body'])['execution_mode']
+    source_ip = event['requestContext']['http']['sourceIp']
```

Comment:
In case you need ;) https://docs.powertools.aws.dev/lambda/python/latest/core/event_handler/api_gateway/#lambda-function-url